core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -601,16 +634,32 @@ public void setClearOnStop(boolean clearOnStop) {
       this.clearOnStop = clearOnStop;
    }
 
+   //TODO should we make a copy for all getters?
```

Comment:
Technically these locks below do nothing at all and we could still get `ConcurrentModificationException`s etc. I would say make a copy like you mentioned. These should be invoked so infrequently it doesn't matter about perf.